integrationtests/src/test/java/io/kroxylicious/proxy/schema/validation/SchemaValidationIT.java

```diff
@@ -0,0 +1,156 @@
+ */
+package io.kroxylicious.proxy.schema.validation;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.InvalidRecordException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.proxy.KroxyConfig;
+import io.kroxylicious.proxy.KroxyConfigBuilder;
+import io.kroxylicious.proxy.VirtualClusterBuilder;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+
+import static io.kroxylicious.proxy.Utils.startProxy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(KafkaClusterExtension.class)
+public class SchemaValidationIT {
+
+    private static final String TOPIC_1 = "my-test-topic";
+    private static final String TOPIC_2 = "my-test-topic-2";
+    public static final String VALID_JSON = "{\"value\":\"json\"}";
+
+    @Test
+    public void testInvalidJsonProduceRejected(KafkaCluster cluster, Admin admin) throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        admin.createTopics(List.of(
+                new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+
+        var config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers())
+                .addNewFilter().withType("ValidJson").endFilter()
+                .build()
+                .toYaml();
+
+        try (var ignored = startProxy(config)) {
+            try (var producer = getProducer(proxyAddress, 0)) {
+                ExecutionException executionException = assertThrows(ExecutionException.class, () -> {
+                    producer.send(new ProducerRecord<>(TOPIC_1, "my-key", "Not Json")).get();
+                });
+                assertTrue(executionException.getCause() instanceof InvalidRecordException);
+            }
+        }
+    }
+
+    @Test
+    public void testPartiallyInvalidJsonProduceRejected(KafkaCluster cluster, Admin admin) throws Exception {
```

Comment:
There's multiple cases you need to test. Here you're testing that topic A can have invalid record and topic B valid ones and that the records for B do get appended, while those for A do not. But there's also the case of different partitions of the same topic. And you also want to assert that when the records for a single partition contain a mix of valid and invalid records then none of them are appended.